designer/server/src/test/scala/pl/touk/nussknacker/ui/api/NodesApiSpec.scala

```diff
@@ -0,0 +1,1093 @@
+                 |            {
+                 |                "name": "id",
+                 |                "expression": {
+                 |                    "language": "spel",
+                 |                    "expression": "'a'"
+                 |                }
+                 |            }
+                 |        ]
+                 |    },
+                 |    "output": "out",
+                 |    "additionalFields": null,
+                 |    "type": "Enricher"
+                 |}""".stripMargin
+            )
+            .when()
+            .post(s"$nuDesignerHttpAddress/api/nodes/${process.name}/additionalInfo")
+            .Then()
+            .statusCode(200)
+            .body(
+              equalsJson(s"""{
+                   |  "content": "\\nSamples:\\n\\n| id  | value |\\n| --- | ----- |\\n| a   | generated |\\n| b   | not existent |\\n\\nResults for a can be found [here](http://touk.pl?id=a)\\n",
+                   |  "type": "MarkdownAdditionalInfo"
+                   |}""".stripMargin)
+            )
+
+          given()
+            .basicAuth("allpermuser", "allpermuser")
+            .and()
+            .jsonBody(
+              s"""{
+                 |    "id": "1",
+                 |    "service": {
+                 |        "id": "otherService",
+                 |        "parameters": []
+                 |    },
+                 |    "output": "out",
+                 |    "additionalFields": null,
+                 |    "type": "Enricher"
+                 |}""".stripMargin
+            )
+            .when()
+            .post(s"$nuDesignerHttpAddress/api/nodes/${process.name}/additionalInfo")
+            .Then()
+            .statusCode(200)
+            .body(
+              equalTo("")
+            )
+        }
+        "return 404 for not existent process" in {
+          val wrongName: String = "wrongProcessName"
+
+          given()
+            .basicAuth("allpermuser", "allpermuser")
+            .and()
+            .jsonBody(
+              s"""{
+                 |    "id": "1",
+                 |    "service": {
+                 |        "id": "otherService",
+                 |        "parameters": []
+                 |    },
+                 |    "output": "out",
+                 |    "additionalFields": null,
+                 |    "type": "Enricher"
+                 |}""".stripMargin
+            )
+            .when()
+            .post(s"$nuDesignerHttpAddress/api/nodes/$wrongName/additionalInfo")
+            .Then()
+            .statusCode(404)
+            .body(
+              equalTo(s"No scenario $wrongName found")
+            )
+        }
+      }
+      "not authenticated should" - {
+        "forbid access" in {
+          createSavedProcess(process, TestCategories.Category1, TestProcessingTypes.Streaming)
+
+          given()
+            .auth()
+            .none
+            .and()
+            .jsonBody(s"""{
```

Comment:
maybe in the case of tests where the request's body is not so important, we should have a `val` with the example body string somewhere and just reference it here? 

In this test, the important thing is the endpoint and credentials (not the body).  The same when we test the case where the process doesn't exist. 

IMO the body should be inlined in tests in which the body is important in case of tested functionality.
WDYT?

Reply:
oh, it seems that I didn't express myself clearly. I didn't mean "local val", but class level val. 

Eg. we have this test:
```scala
    "not authenticated should" - {
        "forbid access" in {
          createSavedProcess(process, TestCategories.Category1, TestProcessingTypes.Streaming)
          val properJson = s"""{
                               |    "id": "1",
                               |    "service": {
                               |        "id": "otherService",
                               |        "parameters": []
                               |    },
                               |    "output": "out",
                               |    "additionalFields": null,
                               |    "type": "Enricher"
                               |}""".stripMargin
          given()
            .noAuth()
            .jsonBody(properJson)
            .when()
            .post(s"$nuDesignerHttpAddress/api/nodes/${process.name}/additionalInfo")
            .Then()
            .statusCode(401)
            .body(
              equalTo("The resource requires authentication, which was not supplied with the request")
            )
        }
      }
```

When you read it, you are interested in checking if request `POST /api/nodes/${process.name}/additionalInfo` cannot be handled when no auth was passed. In this proper example, you are not interested in the body of the request. You don't want to pass to the request body some malformed string, obviously. But as far as the body is valid, it's irrelevant here. 

So, when we define eg.:
```
private val exampleNodesAdditionalInfoRequestBody = 
   s"""{
                               |    "id": "1",
                               |    "service": {
                               |        "id": "otherService",
                               |        "parameters": []
                               |    },
                               |    "output": "out",
                               |    "additionalFields": null,
                               |    "type": "Enricher"
                               |}""".stripMargin
```

somewhere in the Spec class (as far we agree it's not so important, it should probably be placed at the end of the file). 
Then the test will look like this:

```scala
    "not authenticated should" - {
        "forbid access" in {
          createSavedProcess(process, TestCategories.Category1, TestProcessingTypes.Streaming)

          given()
            .noAuth()
            .jsonBody(exampleNodesAdditionalInfoRequestBody)
            .when()
            .post(s"$nuDesignerHttpAddress/api/nodes/${process.name}/additionalInfo")
            .Then()
            .statusCode(401)
            .body(
              equalTo("The resource requires authentication, which was not supplied with the request")
            )
        }
      }
```

Now, it's much more readable, because not important things (emphasizing - in this case!) were moved out of the test.